types/try-to-catch/try-to-catch-tests.ts

```diff
@@ -0,0 +1,9 @@
+import tryToCatch = require("try-to-catch");
+
+const main = async () => {
+    await tryToCatch(() => 5); // $ExpectType [error: null, result: 5]
+
+    await tryToCatch(async () => 5); // $ExpectType [error: null, result: 5]
+
+    await tryToCatch(() => throw Error("error"));  // $ExpectType [error: Error]
```

Comment:
Almost there, just a few fixes:

- Need a body in the function to `throw`
- TypeScript won't be able to refine the type beforehand - the type will be a union

```suggestion
    await tryToCatch(() => 5); // $ExpectType [error: Error] | [error: null, result: number]

    await tryToCatch(async () => 5); // $ExpectType [error: Error] | [error: null, result: number]

    await tryToCatch(() => { throw new Error("error") });  // $ExpectType [error: Error] | [error: null, result: unknown]
```